stdlib/printf.mli

```diff
@@ -188,6 +188,59 @@ val ikbprintf : (Buffer.t -> 'd) -> Buffer.t ->
    @since 4.11
 *)
 
+(** Formatted output functions with heterogeneous list as arguments.
+
+  The following functions behave the same as their non-'l' counter-parts, but
+  receive a heterogeneous list as arguments.
+*)
```

Comment:
Shouldn't this doc contain some motivation for l*-functions?
AFAIU [PR above](https://github.com/ocaml/ocaml/issues/12262#issuecomment-1592989009), it makes error reporting better. Could we mention that?

Reply:
Absolutely! 
I think it is also worth mentioning in the doc that `dprintf` can be derived easily, and kontinuation variants aren't needed because we now know how many arguments the function has, so there is no need for them